js/ssai_pulse.js

```diff
@@ -934,6 +953,55 @@ OO.Ads.manager(function(_, $)
     }, this);
 
     /**
+     * Helper function to get the duration property within the Vast ad object.
+     * @private
+     * @method SsaiPulse#_getDuration
+     * @param {object} vastAdData The Vast ad object
+     * @returns {string} The Vast ad duration time stamp.
+     */
+    var _getDuration = _.bind(function(vastAdData)
+    {
+      var duration = null;
+      if (vastAdData &&
+          vastAdData.linear &&
+          vastAdData.linear.duration)
+      {
+        duration = vastAdData.linear.duration;
+      }
+      return duration;
+    }, this);
+
+    /**
+     * Helper function adjust the duration to a proper value. The priority from which to grab the duration is:
+     * 1. ID3 Tag ad duration - if the value is 0, fall through
+     * 2. VAST XML Response ad duration - if the ad duration is not defined, fall through
+     * 3. DEFAULT_AD_DURATION
+     * @private
+     * @method SsaiPulse#_normalizeDuration
+     * @param {object} id3Object The object containing the ID3 Tag information
+     * @param {object} vastAdData The object containing the parsed Vast ad data
+     * @returns {number} The duration of the ad (in seconds).
+     */
+    var _normalizeDuration = _.bind(function(id3Object, vastAdData)
+    {
+      var duration = FALLBACK_AD_DURATION;
+
+      var vastDuration = _getDuration(vastAdData);
+      vastDuration = adManagerUtils.convertTimeStampToMilliseconds(vastDuration) / 1000;
```

Comment:
Can you add checks to ensure that adManagerUtils.convertTimeStampToMilliseconds(vastDuration) is a number before setting vastDuration to this result?